base_optional_quick_create/model.py

```diff
@@ -50,14 +48,18 @@ def _register_hook(self, cr, ids=None):
                     model_obj.check_quick_create = True
         return True
 
+    def _register_hook(self, cr):
+        self._patch_quick_create(cr, self.search(cr, SUPERUSER_ID, []))
+        return super(IrModel, self)._register_hook(cr)
+
     def create(self, cr, uid, vals, context=None):
-        res_id = super(ir_model, self).create(cr, uid, vals, context=context)
-        self._register_hook(cr, [res_id])
+        res_id = super(IrModel, self).create(cr, uid, vals, context=context)
+        self._patch_quick_create(cr, [res_id])
```

Comment:
> Still it does not seem a good trigger point to do it. If you quick-create a record before slow-creating one, it would still enable it

Why? I don't think this happens.

Consider these are the ```create``` and ```write``` methods of ```ir.model``` model, not of the model itself.

For example: when the ```account.invoice``` (record of ```ir.model```) is modified, setting ```avoid_quick_create``` = True, patch ```name_create``` of ```account.invoice``` model